front/sssscs/src/register/Register.tsx

```diff
@@ -0,0 +1,90 @@
+import { Box, TextField, Button, Typography } from "@mui/material";
+import { AxiosResponse, AxiosError } from "axios";
+import { RegisterService, User, UserCreateDto } from "./RegisterService";
+import { FieldValues, useForm } from "react-hook-form";
+
+const TryRegister = async (data: FieldValues) => {
+    const dto: UserCreateDto = {
+        'email': data['email'],
+        'name': data['name'],
+        'surname': data['surname'],
+        'password': data['password']
+    }
+    RegisterService.register(dto)
+        .then((res: AxiosResponse<User>) => {
+            console.log(res.data);
+            console.log(res.status);
+        })
+        .catch((err : AxiosError) => {
+            console.error(err.response?.data);
+            console.error(err.response?.status);
+        });
+}
+
+const Register = () => {
+    const { register, handleSubmit, formState: { errors } } = useForm({mode: 'onChange'});
+
+    return (
+        <Box component='form' onSubmit={handleSubmit(TryRegister)} sx={{maxWidth: '30rem'}}>
+            <Typography variant='h4' sx={{mb: 2}}>
+                Register
+            </Typography>
+            <TextField
+                sx={{mb: 2}}
+                label="Email"
+                fullWidth
+                {...register('email', {
+                    required: 'Email is required',
+                    pattern: {
+                        value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
+                        message: 'Not a valid email'
+                    }
+                })}
+                error={!!errors['email']}
```

Comment:
How would we go about adding an error message if the email is already taken i.e. after a backend response?